tests/test_models.py

```diff
@@ -857,6 +882,321 @@ def test_string_conversion(self, method):
+            enterprise_customer_user=self.enterprise_customer_user,
+            course_id=self.course_id,
+            consent_granted=False,
+        )
+        learner_data = list(self.active_config.collect_learner_data('dummy-user'))
+        assert len(learner_data) == 0
+
+    @mock.patch('integrated_channels.integrated_channel.models.CourseApiClient')
+    @mock.patch('integrated_channels.sap_success_factors.models.SAPSuccessFactorsEnterpriseCustomerConfiguration')
+    @mock.patch('integrated_channels.integrated_channel.models.GeneratedCertificate')
+    @mock.patch('integrated_channels.integrated_channel.models.CourseKey')
+    @mock.patch('enterprise.lms_api.JwtBuilder', mock.Mock())
+    def test_learner_data_course_incomplete(self, mock_course_key, mock_certificate, mock_configuration,
+                                            mock_course_api):
+        enrollment = EnterpriseCourseEnrollmentFactory(
+            enterprise_customer_user=self.enterprise_customer_user,
+            course_id=self.course_id,
+            consent_granted=True,
+        )
+        # Raise GeneratedCertificate.DoesNotExist
+        mock_course_key.from_string.return_value = None
+        mock_certificate.DoesNotExist = Exception
+        mock_certificate.eligible_certificates.get.side_effect = mock_certificate.DoesNotExist
+
+        # Return instructor-paced course details
+        mock_course_api.return_value.get_course_details.return_value = dict(
+            pacing='instructor',
+        )
+
+        learner_data = list(self.active_config.collect_learner_data('dummy-user'))
+        assert len(learner_data) == 1
+        assert learner_data[0].enterprise_course_enrollment_id == enrollment.id
+        assert learner_data[0].course_id == self.course_id
+        assert not learner_data[0].course_completed
+        assert mock_configuration.get_learner_data.called_with(enterprise_enrollment=enrollment,
+                                                               completed_date=None,
+                                                               grade=None)
+
+    @mock.patch('integrated_channels.sap_success_factors.models.SAPSuccessFactorsEnterpriseCustomerConfiguration')
+    @mock.patch('integrated_channels.integrated_channel.models.GeneratedCertificate')
+    @mock.patch('integrated_channels.integrated_channel.models.CourseKey')
+    @mock.patch('enterprise.lms_api.JwtBuilder', mock.Mock())
+    def test_learner_data_course_complete(self, mock_course_key, mock_certificate, mock_configuration):
+        enrollment = EnterpriseCourseEnrollmentFactory(
+            enterprise_customer_user=self.enterprise_customer_user,
+            course_id=self.course_id,
+            consent_granted=True,
+        )
+
+        # Return a mock certificate
+        mock_course_key.from_string.return_value = None
+        certificate = mock.MagicMock(
+            user=self.user,
+            course_id=self.course_id,
+            grade="A-",
```

Comment:
Refactor moved a lot of this to test_utilities.  Also less mocking was required after the refactor (yay!).

https://github.com/edx/edx-enterprise/commit/7357bba85cbd12d507de2dfa278bec8e29ad4a73#diff-b12dd73705f83d2c7a954a988f607d6fR979